pkg/solver/store/db/db.go

```diff
@@ -159,8 +159,23 @@ func (store *SolverStoreDatabase) GetJobOffers(query store.GetJobOffersQuery) ([
 	if query.NotMatched {
 		q = q.Where("deal_id = ''")
 	}
-	if !query.IncludeCancelled {
-		q = q.Where("state != ?", data.GetAgreementStateIndex("JobOfferCancelled"))
+	if query.Active {
+		q = q.Where("state IN (?)", []uint8{
+			data.GetAgreementStateIndex("DealNegotiating"),
+			data.GetAgreementStateIndex("DealAgreed"),
+			data.GetAgreementStateIndex("ResultsSubmitted"),
+		})
+	}
+	if query.Cancelled {
+		q = q.Where("state IN (?)", []uint8{
+			data.GetAgreementStateIndex("JobOfferCancelled"),
+			data.GetAgreementStateIndex("JobTimedOut"),
+		})
+	} else if !query.IncludeCancelled {
```

Comment:
Does there need to be both a Cancelled and a IncludeCancelled?  Is there a different purpose between the two?

Reply:
Yes, the purpose of `IncludeCancelled` is when you want to query all offers _except_ cancelled ones. The new `Cancelled` query is when you want to retrieve _only_ cancelled offers.

Reply:
Oh actually, maybe we can repurpose `Cancelled`:

- When `Cancelled` is `true`, only return offers in a `JobOfferCancelled` or a `JobTimedOut` state.
- When `Cancelled` is `false`, return all offers not in a `JobOfferCancelled` or a `JobTimedOut` state.

Reply:
Following up, we added a `nil` state where `Cancelled` is not present to query for all job offers.